src/main/java/com/patsi/service/RecipeService.java

```diff
@@ -38,60 +39,68 @@ public UUID registerRecipe(Recipe recipe) {
+
+        try (FileOutputStream outputStream = FileHelper.newFileOutputStream(f)) {
+            outputStream.write(recipeIcon);
+        }
         Recipe recipe = recipeRepository.findById(id).get();
         recipe.setImgURL(id.toString());
         recipeRepository.save(recipe);
     }
 
     //Get Existing Recipe
-    public List<Recipe> getRecipe (){
-        return recipeRepository.findAll();
+    public List<Recipe> getRecipe(String uid) {
+
+        List<Recipe> finalRecipeList = recipeRepository.findAll().stream()
+            .filter(recipe -> recipe.getUid() != null && recipe.getUid().equals(uid))
+            .collect(Collectors.toList());
+        finalRecipeList.forEach((recipe -> {
+            recipe.setUid(null);
+        }));
+        return finalRecipeList;
     }
 
     //Filter Random Recipe List
-    public List<Recipe> filterRandomRecipeList (RecipeType recipeType){
+    public List<Recipe> filterRandomRecipeList(RecipeType recipeType) {
         List<Recipe> tmpRecipeList = recipeRepository.findAll();
         return tmpRecipeList.stream()
-            .filter((recipe)-> recipe.getType() == recipeType)
+            .filter((recipe) -> recipe.getType() == recipeType)
             .collect(Collectors.toList());
     }
 
 
     //Get Random Recipe
-    public Recipe getRandomRecipe(){
+    public Recipe getRandomRecipe() {
         Date d = new Date();
         int currHour = d.getHours();
         List<Recipe> returnList;
-        if (currHour <= 11 && currHour >= 5){
+        if (currHour <= 11 && currHour >= 5) {
             returnList = filterRandomRecipeList(RecipeType.BREAKFAST);
-        }else if (currHour >= 12 && currHour <= 14){
+        } else if (currHour >= 12 && currHour <= 14) {
             returnList = filterRandomRecipeList(RecipeType.LUNCH);
-        }else if (currHour >= 15 && currHour <= 17){
+        } else if (currHour >= 15 && currHour <= 17) {
             returnList = filterRandomRecipeList(RecipeType.AFTERNOON_TEA);
-        }else if (currHour >= 18 && currHour <= 21){
+        } else if (currHour >= 18 && currHour <= 21) {
             returnList = filterRandomRecipeList(RecipeType.DINNER);
-        }else{
+        } else {
             returnList = filterRandomRecipeList(RecipeType.DESSERT);
         }
-        int randomNum = (int)(Math.random()*returnList.size());
-        if (randomNum > 0){
+        int randomNum = (int) (Math.random() * returnList.size());
+        if (randomNum > 0) {
             return returnList.get(randomNum);
-        }else{
+        } else {
             log.info("Checked random recipe");
             List<Recipe> tmpRecipeList = recipeRepository.findAll();
-            if (tmpRecipeList.size() > 0){
-                int randomNumWholeList = (int)(Math.random()*tmpRecipeList.size());
+            if (tmpRecipeList.size() > 0) {
+                int randomNumWholeList = (int) (Math.random() * tmpRecipeList.size());
```

Comment:
same as the previous comment.
1. better create a RandomUtil, then a method wrapping Math.random()
2. create some util method which generate a random int within the given min/max

Reply:
fixed